devmand/gateway/src/devmand/channels/cli/TimeoutTrackingCli.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2019-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+
+#include <devmand/channels/cli/Command.h>
+#include <devmand/channels/cli/TimeoutTrackingCli.h>
+#include <magma_logging.h>
+
+namespace devmand::channels::cli {
+
+using namespace std;
+using namespace folly;
+using devmand::channels::cli::Command;
+
+TimeoutTrackingCli::TimeoutTrackingCli(
+    string _id,
+    shared_ptr<Cli> _cli,
+    shared_ptr<folly::ThreadWheelTimekeeper> _timekeeper,
+    shared_ptr<folly::Executor> _executor,
+    std::chrono::milliseconds _timeoutInterval)
+    : id(_id),
+      cli(_cli),
+      timekeeper(_timekeeper),
+      executor(_executor),
+      timeoutInterval(_timeoutInterval) {
+  shutdown = false;
+}
+
+TimeoutTrackingCli::~TimeoutTrackingCli() {
+  MLOG(MDEBUG) << "[" << id << "] "
+               << "~TTCli";
+  shutdown = true;
+  executor = nullptr;
+  timekeeper = nullptr;
+  cli = nullptr;
+  MLOG(MDEBUG) << "[" << id << "] "
```

Comment:
Not sure if we should log these traces at debug level. Maybe we should define TRACE level ... or remove this logs once we have the code tested.

Reply:
agree we should probably have trace level